cpuid/src/lib.rs

```diff
@@ -154,65 +147,62 @@ pub fn filter_cpuid(
                 entry.edx = 0;
             }
             0xB => {
-                // Hide the actual topology of the underlying host
+                use leaf_0xb::*;
+                //reset eax, ebx, ecx
+                entry.eax = 0 as u32;
+                entry.ebx = 0 as u32;
+                entry.ecx = 0 as u32;
+                // EDX bits 31..0 contain x2APIC ID of current logical processor
+                // x2APIC increases the size of the APIC ID from 8 bits to 32 bits
+                entry.edx = cpu_id as u32;
                 match entry.index {
+                    // Thread Level Topology; index = 0
                     0 => {
-                        // Thread Level Topology; index = 0
-                        if cpu_count == 1 {
-                            // No APIC ID at the next level, set EAX to 0
-                            entry.eax = 0;
-                            // Set the numbers of logical processors to 1
-                            entry.ebx = 1;
-                            // There are no hyperthreads for 1 VCPU, set the level type = 2 (Core)
-                            entry.ecx =
-                                leaf_0xb::LEVEL_TYPE_CORE << leaf_0xb::ecx::LEVEL_TYPE_SHIFT;
-                        } else {
-                            if ht_enabled {
-                                // When HT is enabled, there are 2 logical cores at this level
-                                // To get the next level APIC ID, shift right with 1 because we have
-                                // maximum 2 hyperthreads per core that can be represented by 1 bit.
-                                entry.eax = 1;
-                                entry.ebx = 2;
+                        // To get the next level APIC ID, shift right with at most 1 because we have
+                        // maximum 2 hyperthreads per core that can be represented by 1 bit.
+                        entry
+                            .eax
+                            .write_bits(&eax::APICID, (cpu_count > 1 && ht_enabled) as u32);
+                        // When cpu_count ==1 or HT is disabled, there is 1 logical core at this level
+                        // Otherwise there are 2
+                        entry.ebx.write_bits(
+                            &ebx::NUM_LOGICAL_PROCESSORS,
```

Comment:
Same here, values `1` or `2` are written in `ebx` at `ebx::NUM_LOGICAL_PROCESSORS` offset whereas before they were at offset `0`.